server.go

```diff
@@ -787,3 +793,12 @@ func (w *response) Close() error {
 	}
 	return nil
 }
+
+// TLS implements the TLSResponse.TLS method.
+func (w *response) TLS() *tls.ConnectionState {
+	if v, ok := w.tcp.(*tls.Conn); ok {
+		t := v.ConnectionState()
+		return &t
```

Comment:
why does we need to take the address here?

Reply:
I do not have strong opinions. This was just mimicking the HTTP.Request/HTTP.Response strucs which have a TLS *tls.Connection state attribute, and have this set to nil when not a TLS connection.
I will need to change the signature to  (tls.ConnectionState, err) if we do not want to use address.

Reply:
But ConnectionState() doesn't (for some reason), would be good to just mimic that.

Reply:
I suppose this is because `tls.ConnectionState` is done on a TLS connection, while in our case, we wont know until we can do a type assertion.
I have pushed a diff on top of the current stack. Now, the only way to find it is not a Connectionstate from a TLs connection would be to check that connectionstate.Version == 0